### ISSUE TITLE
Add assertions in tests that would check if message is signed by the right party

### ISSUE THREAD

**Issue by dybi** [golemfactory/concent#562]
Issues comes form comment: https://github.com/golemfactory/concent/pull/513#discussion_r198089467

**TL;DR**
Whenever we create message in tests, we should make sure it is signed by the right party, i.e. 
``` python
sign_message(ack_report_computed_task.report_computed_task, PROVIDER_PRIVATE_KEY)
assert TaskMessage.OWNER_CHOICES.requestor in ack_report_computed_task.EXPECTED_OWNERS
```

The mechanism could be added as a part of `self._get_deserialized_XXX` or can be added manually whenerver message is created.
The mechanizm mentioned above could have a flag to disable the check (i. e. for tests where message is **purposely** signed by the **wrong** party),  sth like:
```python
def _get_deserialized_report_computed_task(
        self,
        ...
        should_check_signature=True

    ):
    if should_check_signature:
        assert TaskMessage.OWNER_CHOICES.provider in report_computed_task.EXPECTED_OWNERS
```

```

**Comment by rwrzesien** [comment]
> assert TaskMessage.OWNER_CHOICES.requestor in ack_report_computed_task.EXPECTED_OWNERS

I think this will not give us any extra information apart from the fact that there is a bug in golem-messages if someone accidentally removes one of the `EXPECTED_OWNERS`. We might have a single test that will go through golem-messages with `EXPECTED_OWNERS` and do sanity check if they are what Concent expects.

And instead of doing that in test methods like `_get_deserialized_report_computed_task` I suggest that, after message has been signed, verify its ownership, using method described in https://github.com/golemfactory/concent/issues/563

**Comment by cameel** [comment]
@rwrzesien @dybi Could you summarize the current status of this task?

**Comment by pawelkisielewicz** [comment]
It was interrupted and never restored. Since this issue we fixed typing in other tasks so this issue can be closed.
